File: src/Application Specific/Classes/Entity Collections/EngineerTimeSheet/EngineerTimeSheetValidator.vb
Namespace Entity

    Namespace EngineerTimeSheets

        Public Class EngineerTimeSheetValidator
            'make sure that contact object is valid
            Inherits BaseValidator

            Public Sub Validate(ByVal oEngineerTimeSheet As EngineerTimeSheet)

                'make sure that contact object is valid
                If oEngineerTimeSheet.Errors.Count > 0 Then
                    Dim de As DictionaryEntry
                    For Each de In oEngineerTimeSheet.Errors
                        Me.AddCriticalMessage(CStr(de.Value))
                    Next
                End If

                If oEngineerTimeSheet.EngineerID = 0 Then
                    AddCriticalMessage("Engineer Missing")
                End If

                If oEngineerTimeSheet.TimeSheetTypeID = 0 Then
                    AddCriticalMessage("Type Missing")
                End If

                If CDate(Format(oEngineerTimeSheet.StartDateTime, "dd/MM/yyyy HH:mm")) >= _
                                    CDate(Format(oEngineerTimeSheet.EndDateTime, "dd/MM/yyyy HH:mm")) Then
                    AddCriticalMessage("End Date/Time must be greater than Start Date/Time Missing")
                End If


                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

            End Sub

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/EngineerTimeSheet/EngineerTimeSheetValidator.cs
using System.Collections;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace EngineerTimeSheets
    {
        // make sure that contact object is valid
        public class EngineerTimeSheetValidator : BaseValidator
        {
            public void Validate(EngineerTimeSheet oEngineerTimeSheet)
            {
                // make sure that contact object is valid
                if (oEngineerTimeSheet.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in oEngineerTimeSheet.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (oEngineerTimeSheet.EngineerID == 0)
                {
                    AddCriticalMessage("Engineer Missing");
                }

                if (oEngineerTimeSheet.TimeSheetTypeID == 0)
                {
                    AddCriticalMessage("Type Missing");
                }

                if (Conversions.ToDate(Strings.Format(oEngineerTimeSheet.StartDateTime, "dd/MM/yyyy HH:mm")) >= Conversions.ToDate(Strings.Format(oEngineerTimeSheet.EndDateTime, "dd/MM/yyyy HH:mm")))
                {
                    AddCriticalMessage("End Date/Time must be greater than Start Date/Time Missing");
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}